content/news/2021-07-15-gsoc-2021-intro.md

```diff
@@ -0,0 +1,190 @@
++++
+title = "Google Summer of Code 2021 Intro"
+date = 2021-07-16
+extra.author = "Adam Blanchet and Devin Ragotzy"
++++
+
+Hello all! This year, Ruma has the pleasure of having two students for [Google
+Summer of Code (GSoC) 2021][gsoc]: Devin and Adam. We'll let them [re]introduce
+themselves.
+
+[gsoc]: https://summerofcode.withgoogle.com/
+
+<hr />
+
+# Ruma API Coverage GSoC
+
+by Adam Blanchet
+
+This year for GSoC, Ruma has two students! I am the "newer" one of the two, and
+my name is Adam Blanchet. I've been at work adding support for more of the
+Matrix Spec in the past few weeks, and will be continuing on doing so for the
+remainder of GSoC.
+
+## Identity Service API
+
+Although I had made progress before the official coding period for GSoC had
+started, I continued with finishing the remaining unimplemented Identity Service
+API endpoints:
+
+- [3PID validity endpoint][ruma/ruma#618]
+- [3PID bind endpoint][ruma/ruma#621]
+- [invitation signing endpoint][ruma/ruma#626]
+- [store invitation endpoint][ruma/ruma#631]
+- [3PID unbind endpoint][ruma/ruma#627]
+
+Apart from the [3PID unbind endpoint][ruma/ruma#627] which is currently blocked,
+this concludes the full coverage of the Identity Service API. This means that
+now projects built upon Ruma will be able to more easily integrate functionality
+with identity servers. This could also enable easier development of an identity
+server written in Rust!
+
+[ruma/ruma#618]: https://github.com/ruma/ruma/pull/618
+[ruma/ruma#621]: https://github.com/ruma/ruma/pull/621
+[ruma/ruma#626]: https://github.com/ruma/ruma/pull/626
+[ruma/ruma#631]: https://github.com/ruma/ruma/pull/631
+[ruma/ruma#627]: https://github.com/ruma/ruma/pull/627
+
+## Implementing MSCs and adding types
+
+I then moved on to a few other miscellaneous issues, to get accustomed to
+working on the other parts of the Ruma project:
+
+- [add client secret and session ID identifier types][ruma/ruma#640]:
+  I found this PR to be more interesting than its title may indicate. Not only
+  did I add two identifier types, but I added a new macro to create validated
+  identifier types.The new macro was also used to greatly simplify the
+  declaration of `ServerName` struct. I really enjoyed working on this PR. It
+  was an occasion to create a new macro to simplify the creation of these and
+  new types.  I had never written macro code before, and it was a good occasion
+  for me to learn something new!
+
+- [add RoomName struct][ruma/ruma#645]:
+  This one did not involve macro code, though similar in principle to the
+  previous PR. It adds a `RoomName` struct to be used in an event, which
+  enforces a maximum length.
+
+- [update endpoints for Blurhash implementation][ruma/ruma#650]:
+  [MSC2488][] adds the use of Blurhashes to the Matrix Spec. Although one
+  `client-api` endpoint in Ruma had already been modified to add Blurhash
+  support, I extended this to all endpoints and events specified in the MSC.
```

Comment:
I think the previous support didn't just affect a single endpoint. Maybe use some more general wording noting previous work on this in Ruma (@Frinksy)

Reply:
```suggestion
  [MSC2488][] adds the use of Blurhashes to the Matrix Spec. Although
  some Blurhash support had been previously added to Ruma, I
  extended this to all endpoints and events specified in the MSC.
```